src/applications/MachineShopSimulator.java

```diff
@@ -88,18 +59,17 @@ private void setMachineChangeOverTimes(SimulationSpecification specification) {
      */
     private void setUpJobs(SimulationSpecification specification) {
         Job theJob;
-        for (int i = 1; i <= specification.getNumJobs(); i++) {
-            int tasks = Task.getNumTasks(specification, i);
-            int firstMachine = 0; // machine for first task
-
-            theJob = new Job(i);
-            for (int j = 1; j <= tasks; j++) {
-                int theMachine = getMachineNumber(specification, i, j);
-                int theTaskTime = getTaskTime(specification, i, j);
-                if (j == 1)
-                    firstMachine = theMachine; // job's first machine
-                theJob.addTask(theMachine, theTaskTime); // add to
-            } // task queue
+        for (int currentJob = 1; currentJob <= specification.getNumJobs(); currentJob++) {
+            int tasks = Job.getNumTasks(specification, currentJob);

```

Comment:
The fact that this is being called in this way implies something bad is happening with regards to the distribution of responsibilities.  You're creating a job variable just to invoke a function from it, but that function doesn't interact with that job variable in any way, shape, or form.  Ideally, it would get the number of tasks from the job it is being invoked from.